README.md

```diff
@@ -4,17 +4,116 @@ This is a service used within the NHS for managing and recording school-aged vac
+
+Changes merged into `main` are deployed to the `qa` and `test` environments for
+testing. When there is a large batch of PRs to merge at once, say after a
+merge-freeze for example, only merge a few at a time to try to make it easier to
+trace any issues that arise during testing.
+
+### Release candidate
+
+Once all the necessary changes are merged and have been tested, create a
+release candidate by creating a tag on the `main` branch. e.g. `v1.0.0-rc1`.
+
+Create a [release in GitHub](https://github.com/nhsuk/manage-vaccinations-in-schools/releases/)
+using this tag, or if one has been created for this version already update the
+tag in it.
+
+At this point the changes in the release will go through the NHS assurance
+processes, and possibly through external testing and assurance. If required it
+can be deployed to the `preview` or `training` environements.
+
+### Releasing
+
+When we are ready to release, update the `release` branch and deploy it to
+production. If there have been no hot-fixes since the last release then this is
+a simple fast-forward merge that has to be done on your localhost (see below for
+how to manage non-fast-forwardable situations):
+
+```shell
+git checkout release
+git pull origin release
+
+# Check that release can be fast-forwarded to the release candidate
+git merge-base --is-ancestor release v1.0.0-rc1 && echo "safe to ff-merge"
+
+git merge --ff-only v1.0.0-rc1
+git tag v1.0.0
+git push --tags origin release
+```
+
+Once this is done, you can deploy `release` to production.
+
+#### When `release` and `main` have diverged
+
+There are cases when `release` won't be fast-forwardable to the release
+candidate on `main`. This will happen when a fix has been applied to the
+`release` branch that circumvented the normal release cycle (AKA hotfix, see
+below).
+
+In these cases the `release` branch will need to be reset to `main`.
+
+```sh
+git checkout release
+git pull origin release
+git reset --hard main
+git tag v2.0.0
+git push --tags origin release
+```
+
+### Deploying with GitHub Actions
+
+Changes to `main` will be automatically deployed to the `qa` and `test`
+environments, where they will be tested before going into a release. Since we
+test changes after integration (merging into `main`), any issues in testing can
+potentially
+
+Once we are ready to do a release we can create a
 
 ## Development
 
+### Branching workflow
+
+We follow the patterns and conventions in [GitHub
+Flow](https://docs.github.com/en/get-started/using-github/github-flow).
+
+Try to only put related changes into a single PR and keep them as small and as focused
+as is reasonable. If you start shaving yaks consider putting these changes into
+a separate PR. Likewise if you find the change you're making is quite large, you
+can spread it across multiple PRs, even if functionality is only partly
+complete any one of them.
+
+Include the Trello card number (e.g. `MAVIS-1000`) in a relevant commit message
+and in the PR description.
```

Comment:
Would it better to recommend linking to the card rather than putting the card number in?

Reply:
Ya, good idea.

I think I was worried that because the link includes (by default) the title of the card, it's 1. a bit messy in the commit message, and 2. could leak information, but I don't know if those are serious enough to really worry about.